### ISSUE TITLE
circular imports and busy __init__.py files

### ISSUE THREAD

**Issue by williballenthin** [fireeye/capa#622]
currently we're getting lucky that things work nicely despite abusing imports. for example, when trying to use MonkeyType:

```
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/user/code/capa-pub/capa/features/freeze.py", line 58, in <module>
    import capa.features.insn
  File "/home/user/code/capa-pub/capa/features/insn.py", line 9, in <module>
    import capa.render.utils
  File "/home/user/code/capa-pub/capa/render/__init__.py", line 11, in <module>
    import capa.rules
  File "/home/user/code/capa-pub/capa/rules.py", line 88, in <module>
    capa.features.insn.API,
AttributeError: module 'capa.features' has no attribute 'insn'
```

I get a similar error in VS Code from pylance:

![image](https://user-images.githubusercontent.com/156560/121432232-a3988c80-c937-11eb-9a60-89ba2b163891.png)

and can reproduce an import issue locally:

![image](https://user-images.githubusercontent.com/156560/121432189-94194380-c937-11eb-941e-daee0c2d96bd.png)

---

`capa/features/__init__.py` defines `Feature`, `String`, `Regex` and some other stuff. also, there are files in the same directory, like `insn.py` and `file.py`.

when you ask to do `import capa.features.insn.API`:
`insn.py` imports `render` imports `rules` references `capa.features.insn.API` which is itself (circular import)  and is further complicated by the fact that python looks in `capa/features/__init__.py` first.

we should fix the circular dependency and the `__init__`/`insn` split.

---

in summary, we have a web of code and dependencies spread among `__init__.py` files and `*.py` files. some of the `*.py` files import from their parents through their dependencies. things somehow work, but it doesn't seem correct, and is preventing me from using MonkeyType easily.

i propose that we refactor our use of `__init__.py` to keep those bare and move code into shared/sub modules. unfortunately, this will be a breaking change, as the location of some of these functions/classes changes. so, we should assess if we can/want to do this before 2.0 and get it in.

---
https://github.com/Khan/style-guides/blob/master/style/python.md#__init__py

**Comment by williballenthin** [comment]
here are the offending files:

```
> fd __init__.py | while read -r LINE; do stat -c "%s %n" "$LINE"; done                                                  feature-447!?
0       capa/__init__.py
9177    capa/features/__init__.py
11467   capa/features/extractors/__init__.py
3549    capa/features/extractors/ida/__init__.py
5878    capa/features/extractors/pefile/__init__.py
1706    capa/features/extractors/smda/__init__.py
2795    capa/features/extractors/viv/__init__.py
0       capa/ida/__init__.py
3467    capa/ida/helpers/__init__.py
4084    capa/ida/plugin/__init__.py
11708   capa/render/__init__.py
```

**Comment by williballenthin** [comment]
# proposed changes:

##  `capa/__init__.py`: none
its already empty.

##  `capa/features/__init__.py`
  - move constants and functions to `capa.features.common`
  - move classes to `capa.features.common`, too

##  `capa/features/extractors/__init__.py`
move logic to `capa/features/extractors/common.py`
or maybe pick a better name, like `feature_extractor.py`?

##  `capa/features/extractors/ida/__init__.py`
move logic to `capa/features/extractors/ida/extractor.py`

##  `capa/features/extractors/pefile/__init__.py`
rename to `capa/features/extractors/pefile.py`

##  `capa/features/extractors/smda/__init__.py`
move logic to `capa/features/extractors/smda/extractor.py`

##  `capa/features/extractors/viv/__init__.py`
move logic to `capa/features/extractors/viv/extractor.py`

##  `capa/ida/__init__.py`: none
its already empty

##  `capa/ida/helpers/__init__.py`
rename to `capa/ida/plugin/helpers.py`

##  `capa/ida/plugin/__init__.py`
im not sure if maybe IDA relies on this location and we cannot change it

##  `capa/render/__init__.py`

  - move "convert" logic to `capa.render.result_document` or similar
  - move `render_vverbose` to `capa.render.vverbose.render`, also `verbose`, `default`, `json`

**Comment by williballenthin** [comment]
the circular import was an issue of:

```
features.insn -> common.Feature -> engine.Result -> rules.index_rules_by_namespace -> features.file -> common.Feature
```

fixed in https://github.com/fireeye/capa/pull/624/commits/527e993bb446686fefb2fb528017ae13710b5dc9 by moving `topologically_order_rules` from `engine` to `rules`.

**Comment by mr-tz** [comment]
@mike-hunhoff do you know if IDA requires `capa/ida/plugin/__init__.py` or if we can move the code?
